### PR TITLE
Set order status

### PR DESCRIPTION
**Pull request by AnthonyNGarcia**
*MERGE IN AFTER REST-FACTOR HAS BEEN MERGED IN FIRST*
-Added new field to Order model: orderStatus
-Added endpoint for updating the status of an order to "Complete" from "In Progress"
-Added two endpoints, for getting orders by either "Complete" or "In Progress"

### CODE REVIEW

File: Pizzeria/src/main/java/com/pizzeria/training/controllers/OrderController.java [hunk 1]
```diff
@@ -62,15 +70,33 @@ public Order updateOrder(@RequestParam ObjectId _id, @RequestBody Order updateOr
 		return orderServ.save(updateOrder);
 	}
 	
+	@PutMapping("/completeOrder")
+	public ResponseEntity<String> completeOrder(@RequestBody String orderId) {
+		return orderServ.updateStatus(new ObjectId(orderId));
+	}
+	
+	@GetMapping("/getOrdersByStatus/{orderStatus}")
```
Comment (Th3Equin0x) [inline Pizzeria/src/main/java/com/pizzeria/training/controllers/OrderController.java:78]: Doesn't seem to match with the HTML standards of endpoints being nouns

File: Pizzeria/src/main/java/com/pizzeria/training/models/Order.java [hunk 2]
```diff
@@ -29,6 +29,7 @@
 	private double cost;
 	private double tip;
 	private boolean setAsFavorite;	//Prevents from being added to the database
+	private String orderStatus;
```
Comment (Th3Equin0x) [inline Pizzeria/src/main/java/com/pizzeria/training/models/Order.java:32]: Potential for an enum due to having limited valid values?

**Review by Th3Equin0x — CHANGES_REQUESTED** [review]
Should name Order status endpoints in accordance with HTML standards, like Ben just refactored them to be (all nouns, for starters). Maybe communicate with Ben about how the endpoints should be written? Maybe they don't even need standalone endpoints, since we have update and getByExample?
Also, is there a reason for the status to be a String instead of enum in the Order model? It seems as if there are limited values we would expect, would would make it perfect for an enum.

File: Pizzeria/src/main/java/com/pizzeria/training/controllers/OrderController.java (hunk 1 above)
Comment (swerdfish) [reply Pizzeria/src/main/java/com/pizzeria/training/controllers/OrderController.java:78]: I would recommend putting the order status as an optional path variable or request param on the existing get /orders endpoint (see lines 55-58).

**Comment by AnthonyNGarcia** [comment]
Renamed/slightly restructured endpoints to more closely resemble HTML standards. Kept endpoint separate from example to provide more specific exception throwing in the case of an invalid enum being provided for a search query (IllegalArgumentException). Converted orderStatus from String to enum.

**Review by Th3Equin0x — APPROVED** [review]

**Pull request merged**